src/main/java/filter/Parser.java

```diff
@@ -173,13 +174,25 @@ private FilterExpression parseQualifier(Token token) {
         qualifierName = qualifierName.substring(0, qualifierName.length() - 1).trim();
 
         QualifierType type = QualifierType.parse(qualifierName).orElse(QualifierType.FALSE);
-        return parseQualifierContent(type, false);
+        if (type == QualifierType.SORT) {
+            //special case for sort
```

Comment:
The original special case was that sorting keys would only parse if preceded by `sort`.

More than one sort qualifier is a semantic error, no need to handle that in the parser. Ideally it could be handled somewhere down the line, but before it gets to UpdateController (which handles the logic of updates, and shouldn't be concerned with checking its inputs).

We should consider what happens when another qualifier is used with sorting keys though, e.g. `id:~id,repo`. Is this a parse error or a semantic error? Maybe it should be the latter, since we have the distinction now?